avogadro/quantumio/nwchemjson.cpp

```diff
@@ -47,112 +45,116 @@ using Core::CrystalTools;
 using Core::Elements;
 using Core::GaussianSet;
 using Core::Molecule;
-using Core::Variant;
 using Core::split;
+using Core::Variant;
 
-NWChemJson::NWChemJson()
-{
-}
+NWChemJson::NWChemJson() {}
 
-NWChemJson::~NWChemJson()
-{
-}
+NWChemJson::~NWChemJson() {}
 
 bool NWChemJson::read(std::istream& file, Molecule& molecule)
 {
-  Value root;
-  Reader reader;
-  bool ok = reader.parse(file, root);
-  if (!ok) {
-    appendError("Error parsing JSON: " + reader.getFormatedErrorMessages());
+  json root;
+  try {
```

Comment:
I am guessing you are using this as you want to get at the parse error object? If you want to do this, then I think we need to do the same for the Chemical JSON format too, probably preferable but I would like us to be consistent.

Reply:
Well, I originally wanted to do it because nlohmann json always throws an exception when there is a parse error, and I didn't want Avogadro2 to crash while parsing an invalid json file.

Since we are catching the exception anyways, I thought it'd be useful to print out the error message.